validator/src/main/java/com/amazon/aoc/services/CloudWatchService.java

```diff
@@ -45,22 +40,40 @@ public CloudWatchService(String region) {
   /**
    * listMetrics fetches metrics from CloudWatch.
    *
-   * @param namespace the metric namespace on CloudWatch
+   * @param namespace  the metric namespace on CloudWatch
    * @param metricName the metric name on CloudWatch
    * @return List of Metrics
    */
   public List<Metric> listMetrics(final String namespace, final String metricName) {
     final ListMetricsRequest listMetricsRequest =
-        new ListMetricsRequest().withNamespace(namespace).withMetricName(metricName);
+            new ListMetricsRequest().withNamespace(namespace).withMetricName(metricName);
     return amazonCloudWatch.listMetrics(listMetricsRequest).getMetrics();
   }
 
+
+  /**
+   * getMetricData fetches the history data of the given metric from CloudWatch.
+   * @param metric metric query object
+   * @param startTime the start timestamp
+   * @param endTime the end timestamp
+   * @return List of MetricDataResult
+   */
+  public List<MetricDataResult> getMetricData(Metric metric, Date startTime, Date endTime) {
+    MetricStat stat = new MetricStat().withMetric(metric).withStat("Average").withPeriod(60);
+    MetricDataQuery query = new MetricDataQuery().withMetricStat(stat).withId("integrationTest");
```

Comment:
small nits, it would be better if we can define magic code `60`, `integrationTest` as vars.

Reply:
Done